### PR TITLE
Zingo sync mempool

### PR DESCRIPTION
**Pull request by Oscar-Pepper**

### CODE REVIEW

**Comment by zancas** [comment]
These changes are incorporated in #1579 .

**Comment by fluidvanadium** [comment]
Why is there not OutgoingTransparentCoin?

**Comment by Oscar-Pepper** [comment]
> Why is there not OutgoingTransparentCoin?

OutgoingSaplingNote and OutgoingOrchardNote come from decrypting the outgoing notes in the transaction but there is no such transparent equivalent.

**Comment by Oscar-Pepper** [comment]
> These changes are incorporated in #1579 .

this PR should be reviewed and merged separately to the removal of getset to keep this PR concise. also there is still ongoing discussion about getset removal

File: libtonode-tests/tests/sync.rs [hunk 1]
```diff
@@ -75,10 +75,10 @@ async fn sync_test() {
     // .await
     // .unwrap();
 
-    increase_server_height(&regtest_manager, 1).await;
-    recipient.do_sync(false).await.unwrap();
-    recipient.quick_shield().await.unwrap();
-    increase_server_height(&regtest_manager, 1).await;
+    // increase_server_height(&regtest_manager, 1).await;
```
Comment (AloeareV) [inline libtonode-tests/tests/sync.rs:78]: Why is the shield test getting commented out? Is this restored by the verification fix followup pr?

File: zingo-sync/src/primitives.rs [hunk 2]
```diff
@@ -284,7 +290,7 @@ pub struct WalletNote<N, Nf: Copy> {
     nullifier: Option<Nf>, //TODO: syncing without nullfiier deriving key
     /// Commitment tree leaf position
```
Comment (AloeareV) [inline zingo-sync/src/primitives.rs:291]: Position is none for mempool and some for confirmed, is this the only source of truth for whether a walletnote is confirmed? If so this needs to be clearer, a comment will probably do the trick,,
```suggestion
    /// Commitment tree leaf position if confirmed, or None for an unconfirmed note
```

File: zingo-sync/src/sync.rs [hunk 3]
```diff
@@ -202,6 +228,102 @@ where
...
+            if mempool_transaction.transparent_coins().is_empty()
+                && mempool_transaction.sapling_notes().is_empty()
+                && mempool_transaction.orchard_notes().is_empty()
+                && mempool_transaction.outgoing_orchard_notes().is_empty()
+                && mempool_transaction.outgoing_sapling_notes().is_empty()
+            {
+                return;
+            }
+
+            // TODO: there is an edge case where txs sent to the mempool that do not return change or create outgoing notes
```
Comment (AloeareV) [inline zingo-sync/src/sync.rs:298]: About this TODO: I think the second step of a send-to-tex doesn't return change, and sends only to transparent. Non-blocking, but worth thinking about.

**Review by AloeareV — APPROVED** [review]

File: libtonode-tests/tests/sync.rs (hunk 1 above)
Comment (Oscar-Pepper) [reply libtonode-tests/tests/sync.rs:78]: this test is just a template test i use to verify things as im developping the PoC. its not a fixed test that will stay in the codebase. integration tests are on the todo lists after PoC is complete

File: zingo-sync/src/primitives.rs (hunk 2 above)
Comment (Oscar-Pepper) [reply zingo-sync/src/primitives.rs:291]: well... the single source of truth for whether a note is confirmed is whether the transaction it belongs to is confirmed. I imagine the position will be None for notes in confirmed transactions when we add support for viewing key only wallets for extra speed up right?

File: zingo-sync/src/sync.rs (hunk 3 above)
Comment (Oscar-Pepper) [reply zingo-sync/src/sync.rs:298]: yes I thought of this and i am already working on this edge case in a following PR. im glad you are thinking deeply about the code in various scenarios though!

**Pull request merged**